app/uk/gov/hmrc/thirdpartydeveloperfrontend/controllers/Details.scala

```diff
@@ -56,8 +56,8 @@ object Details {
     exists: Boolean,
     agreed: Boolean,
     appUsesOldVersion: Boolean,
-    whoAgreed: String,
-    timestampOfWhenAgreed: DateTime
+    whoAgreed: Option[String] = None,
```

Comment:
I wonder if agreed should be optional and be a case class containing who and when.
The view doesn't have to have nasty `.get` anywhere and instead of matching on the boolean and getting optional values. you just match on agreed and safely access the non optional who and when?

Reply:
Nice idea, I'll see what I can do